applications/DEM_application/custom_utilities/move_mesh_utility.cpp

```diff
@@ -0,0 +1,54 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Klauss B Sautter (based on Riccardo Rossi's work)
+//
+// System includes
+
+
+// External includes
+
+
+// Project includes
+#include "move_mesh_utility.h"
+#include "DEM_application_variables.h"
+
+
+namespace  Kratos
+{
+    void MoveMeshUtility::MoveDemMesh(NodesContainerType& rNodes, const bool& rSetDeltaDisplacement) const
+    {
+        KRATOS_TRY;
+
+        KRATOS_ERROR_IF_NOT(rNodes.begin()->SolutionStepsDataHas(DISPLACEMENT_X))
+          << "It is impossible to move the mesh since the DISPLACEMENT var is not in the Model Part"
+          << std::endl;
+
+        const int num_nodes = static_cast<int>(rNodes.size());
+
+        #pragma omp parallel for
+        for(int i = 0; i < num_nodes; i++)
+        {
+            auto it_node = rNodes.begin() + i;
+
+            noalias(it_node->Coordinates()) = it_node->GetInitialPosition().Coordinates();
+            noalias(it_node->Coordinates()) += it_node->FastGetSolutionStepValue(DISPLACEMENT);
+
+            if (rSetDeltaDisplacement)
```

Comment:
Possibly more efficient if you move if-statement outside of the loop.

Reply:
like write a second loop, which is only called if the bool is true?

Reply:
Yes, the argument is that you would then only evaluate the if-clause once instead of `num_nodes` times. Switch statements in loops are bad for performance. But here there might be other switch statements in the functions you call and the time to spawn omp threads 2x might make performance improvement negligible or worse in this case. Perhaps it is best to leave it as it is and only change it if you have profiling data showing there will be a performance improvement.

Reply:
alright then I'll leave it as it is for now

Reply:
I changed it anyway :metal: